tuxemon/core/event/actions/start_battle.py

```diff
@@ -56,31 +56,44 @@ class StartBattleAction(EventAction):
     ]
 
     def start(self):
+        """
+        Assumes single player against single NPC
+        """
         player = self.game.player1
 
         # Don't start a battle if we don't even have monsters in our party yet.
         if not check_battle_legal(player):
-            logger.debug("battle is not legal, won't start")
+            logger.debug("battle is not legal: won't start because we don't have enough monsters in our party")
             return False
 
+        # Fetch the world state
         world = self.game.get_state_name("WorldState")
         if not world:
-            return False
+            logger.debug("battle could not be started because WorldState was not obtained")
+            return False 
+
+        # Fetch the antagonist NPC from the world state
+        npc = world.get_entity(self.parameters.npc_slug)
+
+        # Don't start a battle if we've already battled this npc before
+        if npc.name in player.game_variables['battle_history']:
```

Comment:
I'd prefer to use slugs here because they are less likely to change compared to the name.  If a map designer changes a NPC's name during development, this value will break.  Slugs are our internal IDs for objects and they don't change often.

Reply:
Thanks. Let's go with slug